nwbwidgets/controllers/group_and_sort_controllers.py

```diff
@@ -39,10 +38,10 @@ def __init__(self, dynamic_table: DynamicTable):
 class GroupAndSortController(AbstractGroupAndSortController):
     def __init__(
         self,
-        dynamic_table: DynamicTable,
+        dynamic_table: DynamicTable=None,
```

Comment:
I do not understand this change. How can a GroupAndSortController work without a DynamicTable?

Reply:
if there is no dynamic table present, then essentially this controller will revert to a range_controller for the y axis. This requires the keep_rows (if dt is None) to get the correct range). If keep rows is given with dynamic_table then it applies that to this table instead as per the old design. A better name for 'GroupAndSort' is still pending though

Reply:
I'd rather not do it this way. If a user wants a range controller and does not have a dynamic table to input, they should just use RangeController. If we need a range controller with keep_rows, we should add it to RangeController instead with an arg name like "all_values"

Reply:
I think its lesser on the Modular scale but a little more elegant and saves some lines of code on Classes implementing this controller. One possibility is to rename this to something like `RowsRangeController `and have an arg as `range_controller=False` and `range: list=None` argument (will be provided only if `range_controller=True`) to make things more explicit. 
I feel the group and sorting functionality is a valid subset of `RowsRangeController` (or a similar name) since grouping and sorting process decides the valid range of the rows (if a dynamictable is provided that is) and if not then select all rows.  So I think having it under one class could be feasible at the cost of modularity though. What do you think?

Reply:
what do you think? Would that work? Would it accomplish what you are trying to accomplish here?

Reply:
I don't understand what you mean. I think if a user does not have a DynamicTable and simply wants to control a range, they should use RangeController. If they do have a DynamicTable and want to do grouping, they should use GroupAndSortController.